torchvision/datasets/folder.py

```diff
@@ -17,7 +18,7 @@ def has_file_allowed_extension(filename: str, extensions: Tuple[str, ...]) -> bo
     Returns:
         bool: True if the filename ends with one of given extensions
     """
-    return filename.lower().endswith(extensions)
+    return filename.lower().endswith(extensions if isinstance(extensions, str) else tuple(extensions))
```

Comment:
Do we need this line? `endswith` is supposed to properly support strings and tuples of strings already

Reply:
If you look at the annotations, I annotated `extensions` with `Collection[str]`. This means one could also reasonable call with lists, sets or whatever adheres to the collection protocol. `str.endswith()` however only takes a tuple:

```python
>>> "abc".endswith(["a", "c"])
TypeError: endswith first arg must be str or a tuple of str, not list
```

Thus, we can either limit our input types or simply create a tuple here.

Reply:
Thanks for the details, it's fine to restrict to tuples if it makes the code easier to follow IMO. But I don't want to block either